src/ui/activity-iframe-view.js

```diff
@@ -129,6 +159,14 @@ export class ActivityIframeView extends View {
     return this.promise_;
   }
 
+  /**
+   * Completes the flow.
+   * @return {!Promise}
+   */
+  whenComplete() {
+    return this.whenCompletePromise_;
```

Comment:
Simply return `this.promise_` here. There's no need for a new object, we just needed API separation to preserve the types.

Reply:
Got it. Done